### PR TITLE
CRUD for field of study

### PR DESCRIPTION
**Pull request by DenisZackharov**
### Summary
Same as: ⚠️Parent: #109, with little updates

### How it works

image

### Test plan

List of steps to manually test introduced functionality:

1. Login as director with credentials: `director@example.com` `123456`
2. Click to organization name `School 13`
3. Click`Field of studies`
4. You can create new field of study, edit and destroy existed

1. Login as teacher with credentials: `teacher@example.com` `123456`
2. Click to organization name `School 13`
3. Click `Field of studies`
4. You can see existed field of studies

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests for new functionality
- [ ] My pull request is shorter than 300 lines

### Deploy notes
None.

### CODE REVIEW

File: app/controllers/organizations/field_of_studies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+module Organizations
+  class FieldOfStudiesController < ApplicationController
+    include FlashPreparer
+
+    expose :field_of_study
+    expose :organization, id: -> { params[:organization_id] }
+    expose :field_of_studies, -> { organization.field_of_studies.order(updated_at: :desc) }
+
+    before_action :authorize_field_of_study, only: %i[create update destroy]
```
Comment (egorpavlovs) [inline app/controllers/organizations/field_of_studies_controller.rb:9]: I'm not sure, but I think we can call method like that here:
```suggestion
    before_action { authorize! field_of_study }, only: %i[create update destroy]
```

File: app/interactors/organization_field_of_studies/destroy_record.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    delegate :field_of_study, :organization, to: :context
+
+    def call
+      organization_field_of_study.destroy
+    end
+
+    private
+
+    def organization_field_of_study
+      OrganizationFieldOfStudy.find_by!(field_of_study: field_of_study, organization: organization)
```
[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/interactors/organization_field_of_studies/destroy_record.rb:14]: image

File: app/controllers/organizations/field_of_studies_controller.rb (hunk 1 above)
Comment (DenisZackharov) [reply app/controllers/organizations/field_of_studies_controller.rb:9]: Looks like i can't writ like this, but i can write

```
    before_action only: %i[create update destroy] do
      authorize! field_of_study
    end
```

I don't really like this implementation

File: app/interactors/organization_field_of_studies/destroy_record.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (DenisZackharov) [reply app/interactors/organization_field_of_studies/destroy_record.rb:14]: How you do it?

[the remark this answers is not in the document]
Comment (RasimKhusaenov) [reply app/interactors/organization_field_of_studies/destroy_record.rb:14]: I destroyed record from console before

File: app/interactors/organization_field_of_studies/destroy_record.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module OrganizationFieldOfStudies
+  class DestroyRecord
+    include Interactor
+
+    delegate :field_of_study, :organization, to: :context
+
+    def call
+      organization_field_of_study.destroy
```
[the remark this answers is not in the document]
Comment (DenisZackharov) [reply app/interactors/organization_field_of_studies/destroy_record.rb:8]: No

**Pull request merged**